runtime/util/hshelp.c

```diff
@@ -1701,6 +1701,72 @@ fixJNIRefs(J9VMThread * currentThread, J9HashTable * classPairs, BOOLEAN fastHCR
 }
 
 
+void
+fixMemberNames(J9JavaVM * vm, J9HashTable * classHashTable)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	/* iterate over all objects fixing up vmtarget refs if object is a MemberName */
+	vm->memoryManagerFunctions->j9mm_iterate_all_objects(vm, PORTLIB, 0, fixMemberNamesObjectIteratorCallback, classHashTable);
+}
+
+static jvmtiIterationControl
+fixMemberNamesObjectIteratorCallback(J9JavaVM *vm, J9MM_IterateObjectDescriptor *objectDesc, void *userData)
+{
+	J9HashTable *classHashTable = userData;
+	j9object_t object = objectDesc->object;
+	J9Class *clazz = J9OBJECT_CLAZZ_VM(vm, object);
+
+	if (NULL != classHashTable) {
+		if (clazz == J9VMJAVALANGINVOKEMEMBERNAME(vm) && NULL != J9OBJECT_U64_LOAD(vm->currentThread, object, vm->vmindexOffset)) {
+			j9object_t vmindex = J9OBJECT_U64_LOAD(vm->currentThread, object, vm->vmindexOffset);
+			J9Class *indexClazz = J9OBJECT_CLAZZ_VM(vm, vmindex);
+
+			J9JVMTIClassPair exemplar;
+			J9JVMTIClassPair *result = NULL;
+
+			exemplar.originalRAMClass = J9VMJAVALANGINVOKEMEMBERNAME_CLAZZ(vm->currentThread, object);
```

Comment:
This should be the `examplar.replacementClass.ramClass` as the clazz object should already be updated when this is called. (ie. After `copyPreservedValues` for FSD)

Reply:
So no need for the hashTableFind?

Reply:
you still need to hashTableFind call to check whether the MemberName is referencing target in a class which is being replaced. Is just as this stage, the `MemberName.clazz`'s ref would already be swapped to the replacementClass by earlier call to `copyPreservedValues`

what you are trying to match here should be something like:
```
examplar.replacementClass.ramClass = J9VM_J9CLASS_FROM_HEAPCLASS(vm->currentThread, J9VMJAVALANGINVOKEMEMBERNAME_CLAZZ(vm->currentThread, object));
result = hashTableFind(classHashTable, &exemplar);
```